general/mem_manager.hpp

```diff
@@ -286,12 +279,21 @@ class Memory
    inline void New(int size);
 
    /// Allocate memory for @a size entries with the given MemoryType.
-   /** The newly allocated memory is not initialized, however the given
-       MemoryType is still set as valid.
+   /** The newly allocated memory is not initialized. If @a mt is not a
+       host memory type the memory is set as device valid. Either way
+       the memory is set as host valid.
```

Comment:
Same here -- was this change in behavior necessary?

Reply:
In the master `New_` has a check like `flags |= IsHostMemory(mt) ? Mem::VALID_HOST : Mem::VALID_DEVICE` but since we now pass `h_mt` and `d_mt` to `New_` and we're doing lazy device allocation I changed it to just be `flags |= VALID_DEVICE`.

I'm thinking we should actually change `if (!IsHostMemory(mt)) { flags |= VALID_DEVICE; }` in the implementation of `Memory::New(size, mt)` to something like: `if (!IsHostMemory(mt)) { flags |= VALID_DEVICE & ~VALID_HOST; }` , which will be the same (I think) as master. How does that sound?